File: Excel/532 Challenge.py
import pandas as pd
import numpy as np

path = "532 Grid where each is sum of already filled in surrounding cells.xlsx"
test = pd.read_excel(path, usecols="B:K", skiprows =1 , header=None).values

matrix_size = 10
m = np.zeros((matrix_size, matrix_size))
m[0, 0] = 1

for i in range(matrix_size):
    for j in range(matrix_size):
        if i != 0 or j != 0:
            m[i, j] = np.sum(m[max(i-1, 0):min(i+2, matrix_size),
                               max(j-1, 0):min(j+2, matrix_size)])

print(np.array_equal(m, test))  # True

File: Excel/500-599/532/532 Challenge.R
library(tidyverse)
library(readxl)

path = "Excel/532 Grid where each is sum of already filled in surrounding cells.xlsx"
test  = read_excel(path, range = "B2:K11", col_names = FALSE) %>% as.matrix()

matrix_size <- 10
m <- matrix(0, nrow = matrix_size, ncol = matrix_size)
m[1, 1] <- 1

for (i in 1:matrix_size) {
  for (j in 1:matrix_size) {
    if (i != 1 || j != 1) {
      m[i, j] <- sum(m[pmax(i - 1, 1):pmin(i + 1, matrix_size),
                       pmax(j - 1, 1):pmin(j + 1, matrix_size)])
    }
  }
}

all.equal(m, test, check.attributes = FALSE) # TRUE
